core/src/main/java/org/fao/geonet/resources/CMISConfiguration.java

```diff
@@ -0,0 +1,443 @@
+                parameters.put(SessionParameter.WEBSERVICES_DISCOVERY_SERVICE, getWebservicesDiscoveryService());
+                parameters.put(SessionParameter.WEBSERVICES_MULTIFILING_SERVICE, getWebservicesMultifilingService());
+                parameters.put(SessionParameter.WEBSERVICES_NAVIGATION_SERVICE, getWebservicesNavigationService());
+                parameters.put(SessionParameter.WEBSERVICES_OBJECT_SERVICE, getWebservicesObjectService());
+                parameters.put(SessionParameter.WEBSERVICES_POLICY_SERVICE, getWebservicesPolicyService());
+                parameters.put(SessionParameter.WEBSERVICES_RELATIONSHIP_SERVICE, getWebservicesRelationshipService());
+                parameters.put(SessionParameter.WEBSERVICES_REPOSITORY_SERVICE, getWebservicesRepositoryService());
+                parameters.put(SessionParameter.WEBSERVICES_VERSIONING_SERVICE, getWebservicesVersioningService());
+                repositoryUrl=getWebservicesBaseUrl();
+                break;
+            case BROWSER:
+                parameters.put(SessionParameter.BROWSER_URL, getBrowserUrl());
+                repositoryUrl=getBrowserUrl();
+                break;
+            case ATOMPUB:
+                parameters.put(SessionParameter.ATOMPUB_URL, getAtompubUrl());
+                repositoryUrl=getAtompubUrl();
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported CMIS Binding type '" + getBindingTypeObject().value() + "'.");
+        }
+        parameters.put(SessionParameter.BINDING_TYPE, getBindingTypeObject().value());
+
+        SessionFactory factory = SessionFactoryImpl.newInstance();
+
+        if (repositoryId == null) {
+            if (repositoryName != null) {
+                // Try to find the repository by name.
+                for (Repository repository:factory.getRepositories(parameters)) {
+                    if (repository.getName().equalsIgnoreCase(repositoryName)) {
+                        this.repositoryId = repository.getId();
+                        break;
+                    }
+                }
+            } else {
+                // If repository id is not set then lets get the first repository.
+                // Some product only have one repository.
+                Repository repository = factory.getRepositories(parameters).get(0);
+                this.repositoryId = repository.getId();
+                this.repositoryName = repository.getName();
+            }
+        } else {
+            // Try to find the repository name for the id that we have specified..
+            for (Repository repository : factory.getRepositories(parameters)) {
+                if (repository.getId().equalsIgnoreCase(this.repositoryId)) {
+                    this.repositoryName = repository.getName();
+                    break;
+                }
+            }
+        }
+
+        parameters.put(SessionParameter.REPOSITORY_ID, repositoryId);
```

Comment:
Should be relevant for `null` and throw an Exception?

Reply:
I added a null check